### PR TITLE
Select group when adding credentials from browser extension

### PR DESCRIPTION
**Pull request by varjolintu**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
Allows KeePassXC to set the group when adding new credentials from KeePassXC-Browser. Groups inside Recycle Bin are ignored.

Related KeePassXC-Browser PR: https://github.com/keepassxreboot/keepassxc-browser/pull/396.

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tested manually with KeePassXC-Browser. Tests included in TestGroups.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ I have added tests to cover my changes.

### CODE REVIEW

File: src/core/Group.cpp [hunk 1]
```diff
@@ -785,6 +785,20 @@ Group* Group::findGroupByUuid(const QUuid& uuid)
     return nullptr;
 }
 
+Group* Group::findGroupByName(const QString& name) {
```
[the remark this answers is not in the document]
Comment (varjolintu) [reply src/core/Group.cpp]: It's now returning a list of groups.

File: src/browser/BrowserService.cpp [hunk 2]
```diff
@@ -148,6 +148,29 @@ QString BrowserService::getDatabaseRecycleBinUuid()
...
+    Group* rootGroup = db->rootGroup();
+    if (!rootGroup) {
+        return {};
+    }
+
+    QStringList result;
+    auto children = rootGroup->groupsRecursive(true);
+    for (const auto &c : children) {
+        if (c != db->metadata()->recycleBin()) {
+            result.push_back(c->name());
```
[the remark this answers is not in the document]
Comment (varjolintu) [reply src/browser/BrowserService.cpp]: The comparison is now made with both group name and UUID.

**Comment by varjolintu** [comment]
Some more work is needed. See: https://github.com/keepassxreboot/keepassxc-browser/issues/388

**Comment by varjolintu** [comment]
Added `BrowserService::insideRecycleBin()` for ignoring groups inside Recycle Bin.

File: src/browser/BrowserService.cpp [hunk 3]
```diff
@@ -704,23 +765,38 @@ BrowserService::checkAccess(const Entry* entry,
     return Unknown;
 }
 
+bool BrowserService::insideRecycleBin(Group* group)
```
Comment (droidmonkey) [inline src/browser/BrowserService.cpp]: This should be a member of the group class: ```Group::isRecycled()```

Comment (varjolintu) [reply src/browser/BrowserService.cpp]: This is now done. I also added tests for this to the TestGroups.

**Pull request merged**